src/main/java/seedu/address/logic/commands/FindCommand.java

```diff
@@ -27,9 +28,18 @@ public FindCommand(NameContainsKeywordsPredicate predicate) {
     }
 
     @Override
-    public CommandResult execute(Model model) {
+    public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
-        model.updateFilteredPersonList(predicate);
+
+        // Update the filtered person list based on the predicate
+        try {
+            model.updateFilteredPersonList(predicate);
+        } catch (Exception e) {
+            throw new CommandException("Error occurred while finding persons with specified keywords.");
```

Comment:
![image](https://github.com/user-attachments/assets/cdcea3ea-5fc9-47a5-a84d-ee96cdbd93ab)
Double-check if chunk of code will affect this.